ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Run.hs

```diff
@@ -134,8 +133,7 @@ startChainSyncConnectionThread registry tracer cfg chainDbView fetchClientRegist
               traceUnitWith tracer ("ChainSyncClient " ++ condense srPeerId) "Terminating because of time limit exceeded."
             Nothing ->
               pure ()
-          throwIO exn
-        Right res' -> pure res'
+        Right _ -> pure ()
```

Comment:
What difference does it make not rethrowing?

Reply:
Rethrowing kills the scheduler and the tests. What we want (at least for TimeExceeded) is to register them and then process them in the QuickCheck property.

Reply:
hm, I didn't expect exceptions to kill the tests if thrown inside forked threads.